test/matchers/toEqualParsedQuery.js

```diff
@@ -0,0 +1,73 @@
+const diff = require('jest-diff');
+const N3 = require("n3");
+
+function toEqualParsedQuery(received, expected) {
+    const options = {
+        isNot: this.isNot,
+        promise: this.promise,
+    };
+
+    let message;
+    let pass;
+
+    pass = objectsEqual(received, expected);
+    message = pass
+        ? () =>
+            this.utils.matcherHint('toEqualParsedQuery', undefined, undefined, options) +
+            '\n\n' +
+            `Expected: ${this.utils.printExpected(expected)}\n` +
+            `Received: ${this.utils.printReceived(received)}`
+        : () => {
+            const diffString = diff(expected, received, {
+                expand: this.expand,
+            });
+            return (
+                this.utils.matcherHint('toEqualParsedQuery', undefined, undefined, options) +
+                '\n\n' +
+                (diffString && diffString.includes('- Expect')
+                    ? `Difference:\n\n${diffString}`
+                    : `Expected: ${this.utils.printExpected(expected)}\n` +
+                    `Received: ${this.utils.printReceived(received)}`)
+            );
+        };
+
+
+
+    return {actual: received, message, pass};
+}
+
+
+// Test function which checks if object are equal, keeping into account how N3.DataFactory works
+let objectsEqual = function (received, expected){
+    if (isPrimitive(received) || received === undefined){
+        return received === expected;
+    }
+
+    if (received instanceof N3.DataFactory.internal.Term){
```

Comment:
I suggest making this more generic, and making it work for terms from any data factory. (I foresee we may run into issues in the future because of this)
So for checking if something is a term, you could simply do `if (received.termType)`.

Reply:
I'm fine with this in tests. (We probably want to swap out N3 at a later stage, but not strictly part of this PR.)